src/Tizen.NUI.Components/Controls/AlertDialog.cs

```diff
@@ -17,583 +17,606 @@
+            {
+                Remove(content);
+            }
 
-            defaultContent = messageContent;
+            if (actionContent != null)
+            {
+                Remove(actionContent);
+            }
+
+            if (titleContent != null)
+            {
+                Add(titleContent);
+            }
 
-            return messageContent;
+            if (content != null)
+            {
+                Add(content);
+            }
+
+            if (actionContent != null)
+            {
+                Add(actionContent);
+
+                // FIXME: Now AlertDialog.Padding Top and Bottom increases AlertDialog size incorrectly.
+                //        Until the bug is fixed, padding view is added after action content.
+                if (actionContent == defaultActionContent)
+                {
+                    if (defaultActionContentPadding != null)
+                    {
+                        Add(defaultActionContentPadding);
+                    }
+                }
+            }
         }
 
-        private Button CreateActionButton(string text, EventHandler<ClickedEventArgs> clickedHandler)
+        private TextLabel CreateDefaultTitleContent()
         {
-            if (text == null)
+            return new TextLabel();
+        }
+
+        private TextLabel CreateDefaultContent()
+        {
+            return new TextLabel();
+        }
+
+        private View CreateDefaultActionContent()
+        {
+            return new View()
+            {
+                Layout = new LinearLayout()
+                {
+                    LinearOrientation = LinearLayout.Orientation.Horizontal,
+                },
+            };
+        }
+
+        // FIXME: Now AlertDialog.Padding Top and Bottom increases AlertDialog size incorrectly.
+        //        Until the bug is fixed, padding view is added after action content.
+        private View CreateDefaultActionContentPadding()
+        {
+            var layout = Layout as LinearLayout;
+
+            if ((layout == null) || (defaultActionContent == null))
             {
                 return null;
             }
 
-            //FIXME: Needs to separate GUI implementation codes to style cs file.
-            var actionButton = new Button();
-            actionButton.Text = text;
-            actionButton.Size = new Size(120, 80);
+            Size2D size = null;
 
-            if (clickedHandler != null)
+            if (layout.LinearOrientation == LinearLayout.Orientation.Horizontal)
+            {
+                size = new Size2D(40, defaultActionContent.Size2D.Height);
+            }
+            else
             {
-                actionButton.Clicked += clickedHandler;
+                size = new Size2D(defaultActionContent.Size2D.Width, 40);
             }
 
-            return actionButton;
+            var paddingView = new View()
+            {
+                Size2D = new Size2D(size.Width, size.Height),
+            };
+
```

Comment:
there are warnings in Line 522, 526. 
I think it will be automatically **disposed** by DisposeQueue. but to remove warning and better performance, could you resolve this warnings?
you can ass size.Dispose() just before `return` (line 533)

Reply:
Thank you for your comment.
I have updated the code to use using() to dispose the size variable after it is used.

Reply:
very good!